sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -533,6 +533,18 @@ export class EventHubSender extends LinkEntity {
             return reject(translate(e));
           };
 
+          if (options && options.cancellationToken) {
+            options.cancellationToken.onabort = () => {
+              removeListeners();
+              const desc: string =
+                `[${this._context.connectionId}] Sender "${this.name}" with ` +
+                `address "${this.address}", cannot send the message. opreation ` +
+                `aborted.`;
+              log.error(desc);
+              throw new Error(desc);
```

Comment:
Since rest of the "error" scenarios in this file are using the "reject the promise" pattern,  lets do the same here instead of throwing the error.

`return reject(new Error(desc))`

Also, the current messaging implies that we were not able to send the message, and so it is aborted. Instead the message should be that the send operation has been cancelled by the user

Reply:
I have updated this with https://github.com/Azure/azure-sdk-for-js/pull/3323/commits/710ddd9f339833d4302fc36f14058be34afc038d

Reply:
Reverted `reject(new Error(desc))` -> `throw new Error(err)`

If we reject the promise, it continues the retry operation with this error:
![image](https://user-images.githubusercontent.com/45216704/58738572-33a3b300-83bb-11e9-974c-71c8744f1e50.png)
And this error is retryable, so program continues the operation and doesn't exit.

If we throw the error program works as expcted.